### PR TITLE
🏷️ Add TypeScript typings

### PR DESCRIPTION
**Pull request by ishan-chhabra**
# Description
This adds TypeScript definitions file to `react-otp-input`.
​
## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
​
## Changes
- Add definitions file
​
## Flags
- N/A
​
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    focusInput: (input: number) => void;
+    focusNextInput: () => void;
+    focusPrevInput: () => void;
+    changeCodeAtFocus: (value: string) => void;
+    handleOnPaste: (e: Object) => void;
+    handleOnChange: (e: Object) => void;
+    handleOnKeyDown: (e: Object) => void;
+    handleOnInput: (e: Object) => void;
+    renderInputs: () => any[];
+    render(): any;
```
Comment (apollonian) [inline index.d.ts:26]: I believe we can do away with render in here. Alternatively, make this class extend Component.

Comment (ishan-chhabra) [reply index.d.ts]: @apollonian Improved the code according to the suggestions!

Comment (apollonian) [reply index.d.ts]: Noice!

**Review by apollonian — APPROVED** [review]

**Comment by apollonian** [comment]
@all-contributors please add @ishan-chhabra for code

**Pull request merged**